### PR TITLE
Add ability to specify multiple attributes in sort

### LINKED ISSUE #670: Allow multiple attributes in `sort` filter
`foo|sort(attribute='name,id')` is quite useful in some cases.

The current workaround (using the fact that sorting is stable) looks extreme awful and is not intuitive: `foo|sort(attribute='id')|sort(attribute='name')`

### PR DESCRIPTION
**Pull request by mayuropensource**
Added ability to specify multiple attributes in sort:
e.g. foo|sort(attribute='name,id').

This fixes #670

### CODE REVIEW

File: jinja2/filters.py [hunk 1]
```diff
@@ -84,6 +84,44 @@ def attrgetter(item):
...
+    attributes from a passed object with the rules of the environment.
+    Dots are allowed to access attributes of each attribute.  Integer
+    parts in paths are looked up as integers.
+
+    The value returned by the returned callable is a list of extracted
+    attribute values.
+
+    Examples of attribute: "attr1,attr2", "attr1.inner1.0,attr2.inner2.0", etc.
+    """
+    def _prepare_attribute_parts(attr):
```
Comment (kevin-brown) [inline jinja2/filters.py]: I see that this logic is basically the same logic from `make_attrgetter`. Would it be possible to abstract this out and use it in both places?

That way if we need to adjust this in the future, we won't have to remember to do it in both places.

Comment (artofcomputerprogramming) [reply jinja2/filters.py]: Sure, will do

Comment (mayuropensource) [reply jinja2/filters.py]: Will do

Comment (mayuropensource) [reply jinja2/filters.py]: @kevin-brown pushed a fix.

File: jinja2/filters.py [hunk 2]
```diff
@@ -84,6 +79,45 @@ def attrgetter(item):
...
+        return items
+
+    return attrgetter
+
+
+def _prepare_attribute_parts(attr):
+    if attr is None:
+        return []
+    elif isinstance(attr, string_types):
+        return [int(x) if x.isdigit() else x for x in attr.split('.')]
```
Comment (jab) [inline jinja2/filters.py]: Drive-by review question: Is `x.isdigit()` correct here? What about integer indexes greater than 9? E.g. '10' should be converted to 10, but `'10'.isdigit() == False`, right? What about negative indexes?

Also, right after http://jinja.pocoo.org/docs/2.10/templates/ explains that the dot syntax can be used for attribute *or* item lookup, it says “Additionally, the attr() filter only looks up attributes.” The docstrings here only mention attribute lookup but in fact support item lookup as well, which is inconsistent, is that right?

Comment (ThiefMaster) [reply jinja2/filters.py]: `isdigit()` checks if a string only contains digits

Comment (jab) [reply jinja2/filters.py]: So it does! Thanks for the correction. (Misleading that it isn’t named something plural like `all_digits`.) in any case, it’s still not a sufficient implementation of “is_integer?” if negatives should be supported: `'-1'.isdigit() == False`. Should they be? Docstring need to mention one way or the other?

Comment (jab) [reply jinja2/filters.py]: So it does! Thanks for the correction. (Misleading that it isn’t named something plural like `all_digits`.) in any case, it’s still not a sufficient implementation of “is_integer?” if negatives should be supported: `'-1'.isdigit() == False`. Should they be? Docstring need to mention one way or the other?

Comment (kevin-brown) [reply jinja2/filters.py]: I'm all in favor of keeping this consistent with the old implementation (supporting keys, and not supporting negative indexes) and looking to support future improvements in later pull requests.

Comment (jab) [reply jinja2/filters.py]: Sounds good. If negative indexes should be unsupported now but supported in future, it does seem worth mentioning that they’re not currently supported in the docstring.

And also clarifying in the docstring that this does attribute and item lookup.

**Comment by mayuropensource** [comment]
Hi, 

Is this pr good to go? Do I need to make doc changes to get this pushed? I can open a separate issue to make doc changes.

Thanks,
Mayur

**Comment by mayuropensource** [comment]
@davidism let me know if I need to make any other changes. 

Thanks,
Mayur

**Comment by kevin-brown** [comment]
Thanks for sticking with us through the process!

I've merged this in since the test failures are against Python 3.8 and are unrelated to these changes (there's an issue in coveragepy).

**Pull request merged**